### ISSUE TITLE
r/aws_s3_bucket_object: Add support for S3 Object Lock legal holds

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#7158]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Now that S3 buckets support Amazon S3 Object Lock (https://github.com/terraform-providers/terraform-provider-aws/issues/6634 & https://github.com/terraform-providers/terraform-provider-aws/issues/6872), add support for S3 object [legal holds](
https://docs.aws.amazon.com/AmazonS3/latest/dev/object-lock-overview.html#object-lock-legal-holds) to the `aws_s3_bucket_object` [resource](https://www.terraform.io/docs/providers/aws/r/s3_bucket_object.html) and [data source](https://www.terraform.io/docs/providers/aws/d/s3_bucket_object.html).

**Comment by ewbankkit** [comment]
Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/7159

**Comment by ewbankkit** [comment]
Suggested syntax:

```hcl
resource "aws_s3_bucket_object" "foo" {
  bucket = "my-versioning-bucket"
  key    = "important-data.txt"

  legal_hold {
    status = "ON"
  }
}
```

**Comment by ewbankkit** [comment]
With the new PR I am going with a simplified syntax that is more in line with the S3 API:

```hcl
resource "aws_s3_bucket_object" "example" {
  object_lock_legal_hold_status = "ON"
}
```

**Comment by nywilken** [comment]
The enhancement to the `aws_s3_bucket_object` resource has been merged and will be released with version 2.29.0 of the Terraform AWS provider.
